docs/source/portfolio_optimisation/theory_implied_correlation.rst

```diff
@@ -0,0 +1,164 @@
+
+  - ID of the second element in a cluster
+
+  - Distance between the elements
+
+  - Number of atoms (simple elements from the portfolio and not clusters) inside
+
+ - A linkage object is transformed to reflect the previously created clusters.
+
+ - A transformed local linkage object is added to the global linkage object
+
+ - The distance matrix is adjusted to the newly created clusters - elements that are now in the new clusters are replaced by the clusters in the distance matrix. The distance from the new clusters to the rest of the elements in the distance matrix is calculated as weighted a average of distances of two elements in a cluster to the other elements. The weight is the number of atoms in an element. So, the formula is:
+
+ .. math::
+    DistanceCluster = \frac{Distance_1 * NumAtoms_1 + Distance_2 * NumAtoms_2}{NumAtoms_1 + NumAtoms_2}
+
+ - The linkage object, representing a dendrogram of all elements in a portfolio is the result of the first step of the algorithm. It sequentially clusters two elements together, while measuring how closely together the two elements are, until all elements are subsumed within the same cluster.
+
+2. In the second step, a correlation matrix is derived from the linkage object.
+
+ - One by one, the clusters (each represented by a link in the linkage object) are decomposed to lists of atoms contained in each of the two elements of the cluster.
+
+ - The elements on the main diagonal of the resulting correlation matrix are set to 1s. The off-diagonal correlations between the variables are computed as:
+
+ .. math::
+    \rho_{i,j} = 1 - 2 * d_{i,j}^{2}
+
+3. In the third step, the correlation matrix is de-noised.
+
+ - The eigenvalues and eigenvectors of the correlation matrix are calculated.
+
+ - Marcenko-Pastur distribution is fit to the eigenvalues of the correlation matrix and the maximum theoretical eigenvalue is calculated.
+
+ - This maximum theoretical eigenvalue is set as a threshold and all the eigenvalues above the threshold are shrinked.
+
+ - The de-noised correlation matrix is calculated back from the eigenvectors and the new eigenvalues.
+
+.. tip::
+
+    The algorithm for de-noising the correlation and the covariance matrix is implemented in the RiskEstimators class of the mlfinlab package. It is described in more detail `here <https://github.com/hudson-and-thames/research/blob/master/RiskEstimators/RiskEstimators.ipynb>`_.
+
+
+.. tip::
+
+    This algorithm is described in more detail in the work **Estimation of Theory-Implied Correlation Matrices** *by* Marcos Lopez de Prado `available here <https://papers.ssrn.com/abstract_id=3484152>`_.
+
+Correlation matrix distance
+###########################
+
+The similarity of the Empirical correlation matrix and the Theory-implied correlation matrix can be measured using the correlation matrix distance introduced by Herdin and Bonek `in this paper <https://publik.tuwien.ac.at/files/pub-et_8791.pdf>`_.
```

Comment:
I see you add the link here rather than in the intro. I think its more important to add it to the intro and it is also ok for it to be here and the intro.

Reply:
Added to the intro